src/main/java/org/jabref/gui/JabRefDialogService.java

```diff
@@ -259,6 +260,15 @@ public boolean showConfirmationDialogWithOptOutAndWait(String title, String cont
         return alert.showAndWait();
     }
 
+    @Override
+    public Optional<ButtonType> showCustomButtonDialogWithOptOutAndWait(Alert.AlertType type, String title, String content,
+                                                                        String optOutMessage, Consumer<Boolean> optOutAction,
+                                                                        ButtonType... buttonTypes) {
+        FXDialog alert = createDialogWithOptOut(AlertType.CONFIRMATION, title, content, optOutMessage, optOutAction);
+        alert.getButtonTypes().setAll(buttonTypes);
+        return alert.showAndWait();
+    }
+
```

Comment:
There still seems to be some code duplication with `showConfirmationDialogWithOptOutAndWait`, which is basically now a subset of this method here. Should call this instead.
Also the method argument `type` in the method signature, but is not used in the method.